### DISCUSSION TITLE
Partial Correlation Using Covariance vs OLS methods?

### DISCUSSION THREAD

**Discussion by adam2392** [raphaelvallat/pingouin#273] [Q&A] [answered]
For the partial correlation function in [pingouin](https://github.com/raphaelvallat/pingouin/blob/8825a215f50a21fdaa53aa63fc25a1af44aee7bc/pingouin/correlation.py#L657-L694), I am wondering what the differences are wrt to the method of estimating partial correlation by:

- fitting OLS of Z on X
- fitting OLS of Z on Y
- comparing the two residuals to compute the partial correlation of X and Y given Z

? The function seems to not do this, and I'm wondering if we lose anything or gain anything for doing either procedure.

**Comment by raphaelvallat** [comment, marked as answer]
Hi @adam2392,

Pingouin's partial correlation used to rely on the residuals of linear regression. We changed it in Pingouin 0.4.0. Copy-pasting the changelog below:

> Major refactoring of [pingouin.partial_corr()](https://pingouin-stats.org/generated/pingouin.partial_corr.html#pingouin.partial_corr), which now uses the same method as the R [ppcor](https://cran.r-project.org/web/packages/ppcor/ppcor.pdf) package, i.e. based on the inverse covariance matrix rather than the residuals of a linear regression. This new approach is faster and works better in some cases (such as Spearman partial correlation with binary variables, see [issue 147](https://github.com/raphaelvallat/pingouin/issues/147)). One caveat is that only the Pearson and Spearman correlation methods are now supported in partial/semi-partial correlation.
